### PR TITLE
Proxy hostname fix

### PR DESCRIPTION
**Pull request by MitchellJThomas**
Tested using Vagrant and Berkshelf (with a local version of this cookbook... duh).  Test kitchen work pending...

### CODE REVIEW

File: resources/java.rb [hunk 1]
```diff
@@ -66,7 +66,7 @@
     if new_resource.install_directory != def_install
       options_array << 'INSTALLDIR=' + install_directory
     end
-    if !new_resource.proxy_hostname.empty?
+    if !new_resource.proxy_hostname.to_s.empty?
```
Comment (bespokebob) [inline resources/java.rb]: I think this should be `if !new_resource.proxy_hostname.nil?` or just `if new_resource.proxy_hostname`

File: resources/java.rb [hunk 2]
```diff
@@ -94,7 +94,7 @@
         options_array << '--install-dir ' + install_directory
       end
     end
-    if !proxy_hostname.empty?
+    if !proxy_hostname.to_s.empty?
```
Comment (bespokebob) [inline resources/java.rb]: Same here. This code was previously (before #10) just `if proxy_hostname`

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,10 @@
 # tripwire_agent Cookbook CHANGELOG
 
-## TBD
+## 0.1.7 (2018-02-06)
```
Comment (bespokebob) [inline CHANGELOG.md]: Please leave future release TBD until it's actually released

**Review by bespokebob — CHANGES_REQUESTED** [review]

**Comment by bespokebob** [comment]
See correct syntax for auto-closing issues in commit/PR: https://help.github.com/articles/closing-issues-using-keywords/

File: resources/java.rb (hunk 1 above)
Comment (MitchellJThomas) [reply resources/java.rb]: Not sure if .nil? will capture the empty string case.  I do think just a truthy test like `if new_resource.proxy_name` might work.  A this point I've tested the code above.

File: metadata.rb [hunk 4]
```diff
@@ -4,7 +4,7 @@
 license 'Apache-2.0'
 description 'Installs/Configures tripwire_agent'
 long_description IO.read(File.join(File.dirname(__FILE__), 'README.md'))
-version '0.1.6'
+version '0.1.7'
```
Comment (bespokebob) [inline metadata.rb]: 0.1.6 hasn't been released, so this shouldn't be changing

File: .kitchen.yml [hunk 5]
```diff
@@ -15,7 +15,7 @@ verifier:
   name: inspec
 
 platforms:
-  - name: a_centos_type
+  - name: centos/7
```
Comment (bespokebob) [inline .kitchen.yml]: Please revert this (but good catch on the quotes below)

**Review by bespokebob — CHANGES_REQUESTED** [review]

File: resources/java.rb (hunk 1 above)
Comment (bespokebob) [reply resources/java.rb]: I would prefer the simple test. This seems a little convoluted, and I don't see any other tests in the cookbook that look like this.

**Comment by MitchellJThomas** [comment]
Ok, I was able to get the kitchen tests working at least for Centos/7.

File: .kitchen.yml (hunk 5 above)
Comment (MitchellJThomas) [reply .kitchen.yml]: Good catch... darn it.

**Comment by MitchellJThomas** [comment]
Alright... whew.  At this point I think that this PR fixes issue #16

**Review by bespokebob — APPROVED** [review]

**Pull request merged**